src/Microsoft.Azure.ServiceBus/Primitives/ServiceBusConnection.cs

```diff
@@ -100,5 +100,9 @@ static void CloseConnection(AmqpConnection connection)
 
             return connection;
         }
+
+        /// <summary>Namespace name used for <see cref="ServiceBusConnection"/>.</summary>
+        /// <value>Namespace name.</value>
+        internal string NamespaceName => this.Endpoint.Host.Split(new[] {'.'}, StringSplitOptions.RemoveEmptyEntries)[0];
```

Comment:
I don't know if different domain names are documented anywhere, but the domain names are definitely different. Also, having it as endpoint instead of namespace just makes it 1% more future-proof.. :)